mlflow/store/dynamodb_store.py

```diff
@@ -0,0 +1,707 @@
+            if response['ResponseMetadata']['HTTPStatusCode'] == 200 and 'Items' in response:
+                items += _filter_experiment(response['Items'], view_type, name)
+        return items
+
+    def list_experiments(self, view_type=ViewType.ACTIVE_ONLY):
+        exps = self._list_experiments(view_type=view_type)
+        return [_dict_to_experiment(e) for e in exps]
+
+    def _create_experiment_with_id(self, name, experiment_id, artifact_uri):
+        dynamodb = self._get_dynamodb_resource()
+        table_name = '_'.join([self.table_prefix, DynamodbStore.EXPERIMENT_TABLE])
+        table = dynamodb.Table(table_name)
+        exp = Experiment(
+            experiment_id=experiment_id,
+            name=name,
+            artifact_location=artifact_uri,
+            lifecycle_stage=RunInfo.ACTIVE_LIFECYCLE
+        )
+        response = table.put_item(
+            Item=_entity_to_dict(exp),
+            ReturnConsumedCapacity='TOTAL',
+        )
+        if response['ResponseMetadata']['HTTPStatusCode'] == 200:
+            return experiment_id
+
+    def create_experiment(self, name, artifact_location=None):
+        if name is None or name == "":
+            raise MlflowException("Invalid experiment name '%s'" % name,
+                                  databricks_pb2.INVALID_PARAMETER_VALUE)
+
+        if self._list_experiments(name=name):
+            raise MlflowException("Experiment '%s' already exists." % name,
+                                  databricks_pb2.RESOURCE_ALREADY_EXISTS)
+        # Get all existing experiments and find the one with largest ID.
+        # len(list_all(..)) would not work when experiments are deleted.
+        experiments_ids = [e.experiment_id for e in self.list_experiments(ViewType.ALL)]
+        experiment_id = max(experiments_ids) + 1 if experiments_ids else 0
+        return self._create_experiment_with_id(name, experiment_id, artifact_location)
+
+    def _get_experiment(self, experiment_id):
+        dynamodb = self._get_dynamodb_resource()
+        table_name = '_'.join([self.table_prefix, DynamodbStore.EXPERIMENT_TABLE])
+        table = dynamodb.Table(table_name)
+        response = table.get_item(
+            Key={
+                'experiment_id': experiment_id,
+            },
+            ReturnConsumedCapacity='TOTAL',
+        )
+        if response['ResponseMetadata']['HTTPStatusCode'] == 200 and 'Item' in response:
+            return response['Item']
+
+    def get_experiment(self, experiment_id):
+        """
+        Fetches the experiment. This will search for active as well as deleted experiments.
+
+        :param experiment_id: Integer id for the experiment
+        :return: A single Experiment object if it exists, otherwise raises an Exception.
+        """
+        try:
+            return _dict_to_experiment(self._get_experiment(experiment_id))
+        except Exception:
+            raise MlflowException("Could not find experiment with ID %s" % experiment_id,
```

Comment:
Catching generic Exception is dangerous. Can you test some other way?

Reply:
Fixed this to check if experiment is None before returning.